src/XProfile/VirtoCommerce.XProfile/Authorization/ProfileAuthorizationHandler.cs

```diff
@@ -159,8 +159,15 @@ private static string GetUserId(AuthorizationHandlerContext context)
 
         private async Task<bool> HasSameOrganizationAsync(Contact currentContact, string contactId)
         {
-            var contact = await GetCustomerAsync(contactId) as Contact;
-            return currentContact.Organizations.Intersect(contact?.Organizations ?? Array.Empty<string>()).Any();
+            var result = false;
```

Comment:
https://refactoring.guru/replace-nested-conditional-with-guard-clauses
I do not insist 😃 
``` csharp
   if (currentContact is null)
      return false;

   var contact = await GetCustomerAsync(contactId) as Contact;
   return currentContact.Organizations.Intersect(contact?.Organizations ?? Array.Empty<string>()).Any();
```

Reply:
fixed